communication/communication-core/src/main/java/org/eclipse/jnosql/AbstractGenericType.java

```diff
@@ -0,0 +1,29 @@
+/*
+ *  Copyright (c) 2022 Otávio Santana and others
+ *   All rights reserved. This program and the accompanying materials
+ *   are made available under the terms of the Eclipse Public License v1.0
+ *   and Apache License v2.0 which accompanies this distribution.
+ *   The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ *   and the Apache License v2.0 is available at http://www.opensource.org/licenses/apache2.0.php.
+ *
+ *   You may elect to redistribute this code under either of these licenses.
+ *
+ *   Contributors:
+ *
+ *   Alessandro Moscatelli
+ */
+package org.eclipse.jnosql;
+
+public abstract class AbstractGenericType<T> {
```

Comment:
Please, don't forget the Javadoc explain why this class does exist and the motivation.